### PR TITLE
feat(cientos): 178 v-light-helper

### PR DESCRIPTION
**Pull request by JaimeTorrealba**
This will not close the issue 178

### CODE REVIEW

File: src/core/directives/vLightHelper.ts [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+
+const { logWarning } = useLogger()
+
+export const vLightHelper = {
+  mounted: (el: any) => {
+    if (!el.isLight) {
+      logWarning(`${el.type} is not a light`)
+      return
+    }
+    const currentHelper = helpers[el.type]
```
Comment (alvarosabu) [inline src/core/directives/vLightHelper.ts:12]: Looks good, add a unmounted to dispose the helper as well

**Review by alvarosabu — CHANGES_REQUESTED** [review]

File: src/core/directives/vLightHelper.ts [hunk 2]
```diff
@@ -12,6 +12,15 @@ export const vLightHelper = {
...
     el.parent.add(new currentHelper(el))
   },
+  unmounted: (el: any) => {
+    if (!el.isLight) {
+      logWarning(`${el.type} is not a light`)
+      return
+    }
+    const currentHelper = helpers[el.type]
+    const currentInstance = el.parent.children.find((child: any) => child instanceof currentHelper)
+    el.parent.remove(currentInstance)
```
Comment (alvarosabu) [inline src/core/directives/vLightHelper.ts:22]: @JaimeTorrealba  The helper would not be free from memory unless you call `.dispose()` 
Screenshot 2023-09-16 at 17 04 44

https://threejs.org/docs/#manual/en/introduction/How-to-dispose-of-objects

**Review by alvarosabu — CHANGES_REQUESTED** [review]

File: src/core/directives/vLightHelper.ts (hunk 2 above)
Comment (JaimeTorrealba) [reply src/core/directives/vLightHelper.ts:23]: Good eye there, thanks, I forgot to dispose it.

Fix is on his way :)

**Review by alvarosabu — APPROVED** [review]

File: docs/guide/directives/v-light-helper.md [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+      />
+
+      <TresHemisphereLight
+        v-light-helper
+      />
+  </TresCanvas>
+</template>
+```
+
+Note that you can pass a modifier with the name of a property, for example `v-log:material`, and will log directly the `material` property
```
Comment (alvarosabu) [inline docs/guide/directives/v-light-helper.md:29]: Copy paste ;) @JaimeTorrealba

**Review by alvarosabu — CHANGES_REQUESTED** [review]

**Comment by alvarosabu** [comment]
@JaimeTorrealba dunno if it's possible but add an update call `helper.update()` everytime there is a change in the light. Otherwise the helped will remind in the same place if you move the light or change the color

**Comment by JaimeTorrealba** [comment]
Interesting, I could use it with the renderLoop, or in the update directive cycle

1. With the renderloop is the same as the request animation frame, will not be performance problems right?
2. Option 2 it will only trigger when a reactive property (Vue) of the element changes (which is not always the case)

**Comment by alvarosabu** [comment]
> Interesting, I could use it with the renderLoop, or in the update directive cycle
> 
> 1. With the renderloop is the same as the request animation frame, will not be performance problems right?
> 2. Option 2 it will only trigger when a reactive property (Vue) of the element changes (which is not always the case)

I will go for the second option since the update is not like the `controls.update()` which needs to be inside of the renderLoop

**Review by alvarosabu — APPROVED** [review]

**Pull request merged**